src/Raven.Server/Rachis/CandidateAmbassador.cs

```diff
@@ -100,110 +105,68 @@ private void Run()
                         }
                         Status = AmbassadorStatus.Connected;
                         StatusMessage = $"Connected to {_tag}";
-                        using (var connection = new RemoteConnection(_tag, _engine.Tag, _conenctToPeer))
+
+                        Connection = new RemoteConnection(_tag, _engine.Tag, stream);
+                        using (_engine.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
                         {
-                            try
+//                            try
+//                            {
+//                                _engine.AppendStateDisposable(_candidate, _connection);
```

Comment:
Why is this commented?

Reply:
It should be removed. 
Basically it says to dispose the connection once we change our state. We want to avoid this in order to reuse the connection.